pkgs/os-specific/linux/morse-driver/default.nix

```diff
@@ -0,0 +1,65 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  kernel,
+}:
+
+stdenv.mkDerivation {
+  pname = "morse-driver";
+  version = "1.14.1-${kernel.version}";
+
+  src = fetchFromGitHub {
+    owner = "MorseMicro";
+    repo = "morse_driver";
+    rev = "33f0092d5c859d8589b99c6e9a77abc58c093648";
+    hash = "sha256-VNw5OdxkP8GltYU2kDYdBUGKWIlPhr3tkzqtW+3MigE=";
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    ./0001-fix-resolve-build-issues-with-latest-Linux-kernels-6.patch
+  ];
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+  hardeningDisable = [ "pic" ];
+
+  makeFlags = [
+    "ARCH=${kernel.arch or stdenv.hostPlatform.linuxArch}"
+    "KERNEL_SRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+
+    # --- Morse-specific Kconfig options ---
+    "CONFIG_WLAN_VENDOR_MORSE=m"
+    "CONFIG_MORSE_SDIO=y"
+    "CONFIG_MORSE_SDIO_ALIGNMENT=4"
+    "CONFIG_MORSE_USER_ACCESS=y"
+    "CONFIG_MORSE_VENDOR_COMMAND=y"
+    "CONFIG_MORSE_COUNTRY=US"
+    "CONFIG_MORSE_DEBUG_MASK=4"
+    "CONFIG_MORSE_SDIO_RESET_TIME=400"
+    "CONFIG_MORSE_POWERSAVE_MODE=0"
+    "CONFIG_MORSE_WATCHDOG_RESET_DEFAULT_DISABLED=y"
+  ]
+  ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
+    "CROSS_COMPILE=${stdenv.cc.targetPrefix}"
+  ];
+
+  installPhase = ''
+    mkdir -p "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+    find . -name "*.ko" -exec cp {} $out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/ \;
```

Comment:
Does it build modules into subdirectories?

Reply:
fixed with https://github.com/NixOS/nixpkgs/pull/458358/commits/85b080e4b5eb9ce52fa3241490fe5f7e2891fb20